src/main/java/de/tum/in/www1/artemis/web/rest/TutorParticipationResource.java

```diff
@@ -0,0 +1,172 @@
+        this.tutorParticipationService = tutorParticipationService;
+        this.exerciseService = exerciseService;
+        this.courseService = courseService;
+        this.userService = userService;
+        this.exampleSubmissionRepository = exampleSubmissionRepository;
+        this.exampleSubmissionService = exampleSubmissionService;
+    }
+
+    /**
+     * POST /exercises/:exerciseId/tutorParticipations : start the "id" exercise for the current user.
+     *
+     * @param exerciseId the id of the exercise for which to init a tutorParticipations
+     * @return the ResponseEntity with status 200 (OK) and with body the exercise, or with status 404 (Not Found)
+     */
+    @PostMapping(value = "/exercises/{exerciseId}/tutorParticipations")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<TutorParticipation> initTutorParticipation(@PathVariable Long exerciseId) throws URISyntaxException {
+        log.debug("REST request to start tutor participation : {}", exerciseId);
+        Exercise exercise = exerciseService.findOne(exerciseId);
+        Course course = exercise.getCourse();
+        User user = userService.getUserWithGroupsAndAuthorities();
+
+        if (!courseService.userHasAtLeastTAPermissions(course)) {
+            return forbidden();
+        }
+
+        TutorParticipation existingTutorParticipation = tutorParticipationService.findByExerciseAndTutor(exercise, user);
+        if (existingTutorParticipation != null && existingTutorParticipation.getId() != null) {
+            // tutorParticipations already exists
+            return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert("tutorParticipations", "tutorParticipationAlreadyExists", "There is already a tutorParticipations for the given exercise and user.")).body(null);
+        }
+        TutorParticipation tutorParticipation = tutorParticipationService.createNewParticipation(exercise, user);
+        return ResponseEntity.created(new URI("/api/exercises/" + exerciseId + "tutorParticipations/" + tutorParticipation.getId()))
+            .body(tutorParticipation);
+    }
+
+    /**
+     * POST /exercises/:exerciseId/tutorParticipations/:participationId/exampleSubmission: add an example submission to the tutor participation
+     *
+     * @param exerciseId      the id of the exercise of the tutorParticipations
+     * @return the ResponseEntity with status 200 (OK) and with body the exercise, or with status 404 (Not Found)
+     */
+    @PostMapping(value = "/exercises/{exerciseId}/exampleSubmission")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<TutorParticipation> addExampleSubmission(@PathVariable Long exerciseId, @RequestBody ExampleSubmission exampleSubmission) throws URISyntaxException {
+        log.debug("REST request to add example submission to exercise id : {}", exerciseId);
```

Comment:
This REST call is rather long and includes a lot of logic. I suggest putting this logic into a service call

Reply:
How can I do that? There are multiple exit points here, since there are a lot of different checks, and every exit point is a different response entity

Reply:
Use Exceptions and either handle them here in the Resource or let them be handled automatically (talk to Max. Jelinek)

Reply:
Done, at the moment I am using standard exception, when the PR 160 will be merged I will move to our new exception system, I have left a todo for me